lib/src/client/channel.dart

```diff
@@ -39,6 +41,12 @@ abstract class ClientChannel {
   /// Initiates a new RPC on this connection.
   ClientCall<Q, R> createCall<Q, R>(
       ClientMethod<Q, R> method, Stream<Q> requests, CallOptions options);
+
+  /// Stream of connection state changes
+  /// 
+  /// This returns a broadcast stream that can be listened to for connection changes.
+  /// Note: on web channels, this will not yield any values.
```

Comment:
Theoretically we could provide implementation for the web channels as well (as underlying XHR transport has `onReadyStateChange`). One source of complexity here would be that on Web one `ClientChannel` corresponds to many connections. You don't need to fix it now but we should write code in a generic fashion (e.g. have `ClientConnection.onStateChange` as suggested) so that we can easily extend it later if needed.